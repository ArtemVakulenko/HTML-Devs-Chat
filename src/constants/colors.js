export const colors = {
  dark: {
    main: '#0E1621',
    secondary: '#17212B',
    input: '#242F3D',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#fff',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#0E1621',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#488ccf',
  },
  light: {
    main: '#FFFFFF',
    secondary: '#ECECEC',
    input: '#D1D1D1',
    header: '#2B5278',
    button: '#2B5278',
    message: '#2B5278',
    buttonText: '#fff',
    notification: '#5C9DDD',
    textError: '#ff0000',
    color: '#000',
    font: '#ffffff',
    placeholder: '#55667B',
    modal: '#FFFFFF',
    modalWrapper: 'rgba(0, 0, 0, 0.5)',
    selectColor: '#000',
    guestName: '#2B5278',
  },
};
