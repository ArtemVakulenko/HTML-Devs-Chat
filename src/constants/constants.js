export const langs = {
  arabian: 'ar',
  russian: 'ru',
  english: 'en',
};

export const cookieAge = 20000;
