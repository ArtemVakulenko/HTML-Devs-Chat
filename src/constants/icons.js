export const icons = {
    changeThemeIcon: '../assets/img/changeThemeIcon.png',
    defaultUserIcon: '../assets/img/defaultUserIcon.png',
    dropDownIcon: '../assets/img/dropDownIcon.png',
    logoIcon: '../assets/img/logoIcon.png',
    logOutIcon: '../assets/img/logOutIcon.png',
    settingIcon: '../assets/img/settingsIcon.png',
    closeModalDark: '../assets/img/closeModalDark.png',
    sendMessage: '../assets/img/sendMessage.png',
    showPass: '../assets/img/view.png',
    hidePass: '../assets/img/hide.png',
    themeDark: '../assets/img/moon.png',
    themeLight: '../assets/img/sun.png',
};
