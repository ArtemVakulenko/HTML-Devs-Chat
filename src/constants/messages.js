export const MESSAGES_KEYS = {
  invalidLogin: 'invalidLogin',
  invalidPassword: 'invalidPassword',
  invalidConfirm: 'invalidConfirm',
  userIsReg: 'userIsReg',
  notExist: 'notExist',
  wrongPass: 'wrongPass',
  abstract: 'something went wrong',
  successReg: 'successReg',
  successAuth: 'successAuth',
  unauth: 'unAuth',
};
