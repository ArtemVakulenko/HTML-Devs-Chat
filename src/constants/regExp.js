export const LOGIN = new RegExp(/^[a-z0-9]{3,18}/, 'i');
export const PASSWORD = new RegExp(/^[a-z0-9!@$^."â„–;%:?*\(\)-_=+]{5,20}$/, 'i');
