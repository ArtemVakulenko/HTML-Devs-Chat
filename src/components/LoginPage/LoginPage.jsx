import React, { useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { MESSAGES_KEYS } from 'constants';
import { NotificationManager } from 'react-notifications';
import { useTranslation } from 'react-i18next';
import Title from '../_common/Title';
import Button from '../_common/Button';
import AuthInput from '../_common/AuthInput';
import CustomLink from '../_common/CustomLink';
import { StLoginContainer, StLoginView } from './styled';
import { useTheme } from '../hooks';

const LoginPage = ({ message, signInRequest, setErrorMessage }) => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const redirect = useMemo(() => {
    if (message === MESSAGES_KEYS.successAuth) {
      return <Redirect to="/main" />;
    }
    return null;
  }, [message]);
  useEffect(() => {
    if (message === MESSAGES_KEYS.successAuth) {
      return NotificationManager.success(t(message), t('Success'), 3000);
    }
    if (message && message !== MESSAGES_KEYS.successReg) {
      NotificationManager.error(t(message), t('Error'), 3000);
      setErrorMessage('');
    }
  }, [message]);
  useEffect(() => {
    return () => setErrorMessage('');
  }, []);
  const onKeyDown = (e) => {
    if (e.key === 'Enter') signInRequest();
  };
  return (
    <StLoginContainer onKeyDown={onKeyDown}>
      <StLoginView colors={colors} theme={theme}>
        <Title title="authorization" />
        <AuthInput type="login" page="login" />
        <AuthInput type="password" page="login" />
        <Button onClick={signInRequest} value="LoginBtn" />
        <CustomLink path="/registration" title="toReg" />
      </StLoginView>
      {redirect}
    </StLoginContainer>
  );
};

export default LoginPage;

LoginPage.propTypes = {
  message: PropTypes.string.isRequired,
  signInRequest: PropTypes.func.isRequired,
  setErrorMessage: PropTypes.func.isRequired,
};
