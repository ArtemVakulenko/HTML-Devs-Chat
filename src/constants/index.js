import { colors } from './colors';
import { icons } from './icons';
import { MESSAGES_KEYS } from './messages';
import { url } from './url';
import { LOGIN, PASSWORD } from './regExp';
import { langs, cookieAge } from './constants';

export { url, colors, icons, MESSAGES_KEYS, LOGIN, PASSWORD, langs, cookieAge };
