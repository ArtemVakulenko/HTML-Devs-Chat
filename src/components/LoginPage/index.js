import { connect } from 'react-redux';
import { signInRequest, setErrorMessage } from 'store/auth/authActions';
import { getErrorMessage } from 'store/auth/selectors';
import LoginPage from './LoginPage';

const mapStateToProps = (state) => ({
  message: getErrorMessage(state),
});

const mapDispatchToProps = {
  signInRequest,
  setErrorMessage,
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
