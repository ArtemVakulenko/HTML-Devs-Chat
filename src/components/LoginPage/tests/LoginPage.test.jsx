import React from 'react';
import configureStore from 'redux-mock-store';
import { NotificationManager } from 'react-notifications';
import LoginPage from '../LoginPage';
import { shallowSmart, mountSmart } from '../../../__tests__/testHelper';
import { MESSAGES_KEYS } from '../../../constants';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({
  auth: {
    auth: {
      login: '',
      password: '',
      confirm: '',
    },
    login: {
      login: '',
      password: '',
    },
    errorMessage: '',
  },
});
jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('LoginPage', () => {
  describe('message !success', () => {
    const props = {
      message: 'MockValue',
      signInRequest: jest.fn(),
      setErrorMessage: jest.fn(),
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<LoginPage {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have StLoginContainer', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('styled__StLoginContainer')).toHaveLength(1);
    });
    it('Should have StLoginView', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('styled__StLoginView')).toHaveLength(1);
    });
    it('Should have Title', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('Title')).toHaveLength(1);
    });
    it('Should have AuthInput', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('AuthInput')).toHaveLength(2);
    });
    it('Should have Button', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('Button')).toHaveLength(1);
    });
    it('Should have CustomLink', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('CustomLink')).toHaveLength(1);
    });
    it('Shouldn`t have redirect', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('Redirect')).toHaveLength(0);
    });
    it('Should have notification error', () => {
      mountSmart(<LoginPage {...props} />, store);
      expect(NotificationManager.error).toHaveBeenCalled();
    });
  });
  describe('message success', () => {
    const props = {
      message: MESSAGES_KEYS.successAuth,
      signInRequest: jest.fn(),
      setErrorMessage: jest.fn(),
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<LoginPage {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
    it('Should have Redirect', () => {
      const component = mountSmart(<LoginPage {...props} />, store);
      expect(component.find('Redirect')).toHaveLength(1);
    });
    it('Should have notification error', () => {
      mountSmart(<LoginPage {...props} />, store);
      expect(NotificationManager.success).toHaveBeenCalled();
    });
  });
});
