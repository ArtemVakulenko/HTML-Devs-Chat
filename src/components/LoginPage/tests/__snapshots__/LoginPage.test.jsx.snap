// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage message !success Should match snapshot 1`] = `"<div class=\\"styled__StLoginContainer-hSVgSI hxbETG\\"><div class=\\"styled__StLoginView-bXdzsa hcjlfy\\"><h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\"></h1><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"text\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/></div><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><img src=\\"../assets/img/view.png\\" class=\\"styled__StHidePass-kZfxvt kfuTnG\\"/></div><button class=\\"styled__StButton-iwFefC fohYxd\\"></button><a href=\\"/registration\\"></a></div></div>"`;

exports[`LoginPage message success Should match snapshot 1`] = `"<div class=\\"styled__StLoginContainer-hSVgSI hxbETG\\"><div class=\\"styled__StLoginView-bXdzsa hcjlfy\\"><h1 class=\\"styled__AppTitle-bMNsHC jWDbjD\\"></h1><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"text\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/></div><div class=\\"styled__AuthInputContainer-iSEnWy kChWax\\"><p class=\\"styled__AuthLabel-jINklJ iykkVO\\"></p><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW bBpwQJ\\"/><img src=\\"../assets/img/view.png\\" class=\\"styled__StHidePass-kZfxvt kfuTnG\\"/></div><button class=\\"styled__StButton-iwFefC fohYxd\\"></button><a href=\\"/registration\\"></a></div></div>"`;
